docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,228 @@ The remaining points can happen any time after 1) is complete although note that
 ### Key Functionality
 
 1. Installation of suites on platforms and support for moving the share, work &
-   log directories to different locations. Currently this is done by
-   `rose suite-run` but we will include this in the
+   log directories to different locations. This will be included in the
    [platforms support](proposal-platforms.md#enhancements-to-support-rose-suite-run-functionality).
 
-2. On start up and reload, install the suite from the source location (working
-   copy) to the run location (`cylc-run`).
-   * Currently this functionality assumes a shared filesystem but we should
-     support workflow server hosts with separate filesystems (future
-     enhancement?).
-   * Needs to create the run time directory structure which has been configured
-     for the `localhost` platform.
+2. Support installation of a suite from the source location (working copy) to
+   the run location (`cylc-run`).
+   * This functionality assumes the source location (working copy) and the run
+     location (`cylc-run`) are both accessible on the host where the command is
+     run.
+   * The installation needs to create the run time directory structure which has
+     been configured for the `localhost` platform.
+   * Need to be able to update from the source location as part of reload.
 
-3. On start up, housekeep / archive logs and other items generated by the suite.
+3. On start up, archive any previous logs in a tar.gz.
+   * Propose to stop supporting this since, in most cases, we will use a
+     separate run directory each time a suite is installed.
+   * Continue to support a timestamped log dir for the special occasions where
+     we re-use an existing run directory (see the discussion in the CLI changes
+     below).
 
 4. On start up and reload, validate suite (using `--strict`).
    * Hopefully this can become a standard part of suite start-up rather than a
      separate step. This was difficult previously because cylc loads the suite
      after it has been detached?
-   * Consider making all validation strict? If we want really want to allow
-     naked dummy tasks in some circumstances then we would be better to add a
-     suite configuration for this.
+   * Consider making all validation strict?
+     * If we really want to allow naked dummy tasks in some circumstances then
+       we would be better to add a suite configuration for this?
+     * Note: we don't allow you to inherit from an undeclared ("naked") family
+       so why do we allow you to run an undeclared task?
```

Comment:
```suggestion
     * If we really want to allow naked tasks in some circumstances then
       we would be better to add a suite configuration for this?
```

A facetious correction, however, naked tasks are not dummy tasks, they still run and just because they are not explicitly defined does not mean they haven't been configured.

Consider this common pattern:

```ini
[runtime]
    [[root]]
        script = rose task-run
```

So-called "naked" tasks are a valid feature of Cylc. There can be cases where explicitly defining runtime tasks may be inconvenient, e.g. for highly programatic suites where obtaining a full list of tasks present in the graph may require building dependencies in Python and writing them in Jinja2 (the other option being introspection 🤮).

I feel that this is somewhat over-opinionated Rose logic I would like to loose. Propose printing a warning listing the first `N` naked tasks. Note that this kind of check would normally be performed by a static code analyser and not provide a barrier to execution.

Reply:
Personally I think there are very few cases where you want to run a task which is purely defined by the root environment. If we allow workflows to run with naked dummy tasks (something `rose suite-run` prevents) I fear we will end up with lots more mistakes and lots of wasted time / resources.

Reply:
> A facetious correction, however, naked tasks are not dummy tasks, they still run and just because they are not explicitly defined does not mean they haven't been configured.

Fair point but the word "dummy" is somewhat semantically fraught.  I've always meant "dummy task" as a real task that has had its guts replaced by `/usr/bin/true` or whatever.  As opposed to a "simulated task" which is what you get in simulation mode. A "naked dummy task" is one that had no guts defined in the first place, but is given a `/usr/bin/true`.

Reply:
I think the one good use for naked dummy tasks is for experimenting with Cylc scheduling (in any number of ways) when you don't care about the task content.  Just write some graph structure and run to see if it behaves as you expect or whatever.  This would be significantly (although not horrifically) more painful if we were forced to write a `[runtime]` subsection for every task.  I'd be open to requiring an explicit configuration setting to allow it though (`allow naked dummy tasks = True` or whatever).  Also open to considering a better name for these things, if we want to go there.

Reply:
The new config setting could come with a validation warning: "WARNING allowing naked dummy tasks is dangerous, please disallow it before running this flow in production" or something.

Reply:
> I think the one good use for naked dummy tasks is for experimenting with Cylc scheduling

Like for instance, the Cylc Tutorial!

(Users hit more issues in the scheduling section than the runtime section since all of the workflow complexity is there so it makes sense to tackle the graph first.)

Reply:
Putting it into a new config would kinda kill this use by making it clunky.

Reply:
So, maybe it is sufficient to allow naked dummy tasks (because sometimes it is very convenient) but WARN if any are detected that it is a dangerous practice in production mode.

Reply:
T'would be my preference.

Users need to get more adept at reading warnings. We need to get better at simplifying output to encourage this practice.

Reply:
I think "play around using dummy tasks" undersells something that we should encourage users to do a lot more of - instead of trying to get some new (to them) aspect of Cylc working by experimenting with their real suites (which is slow and hard to debug etc.). 

But fair enough; new Issue: https://github.com/cylc/cylc-flow/issues/3866